src/pudl/extract/gridpathratk.py

```diff
@@ -0,0 +1,85 @@
+"""Extract GridPath RA Toolkit renewable energy generation profiles from CSVs.
+
+There are several time series available from the GridPath Resource Adequacy Toolkit.
+
+Solar:
+
+* Hourly Solar by Generator
+* Hourly Solar by Zone
+* Hourly Solar by Zone (Extended)
+* Solar Aggregations
+
+Wind:
+
+* Hourly Wind by Plant
+* Hourly Wind by Zone
+* Hourly Wind by Zone (Extended)
+* Wind Aggregations
+
+Thermal Derates:
+
+* Hourly Thermal Derates by Generator
+* Hourly Thermal Derates by Generator (Extended)
+
+Misc:
+
+* Daily Weather Data
+
+"""
+
+from io import BytesIO
+from pathlib import Path
+
+import pandas as pd
+from dagster import AssetsDefinition, Output, asset
+
+# TODO: Reorganize the archive to use asset names for partitions directly
+GRIDPATHRATK_PARTS = {
+    "hourly-solar-zonal-capacity-synth": {
+        "table_name": "aggregated_extended_solar",
+    },
+    "hourly-wind-zonal-capacity-synth": {
+        "table_name": "aggregated_extended_wind",
+    },
+}
+
+
+def raw_gridpathratk_profile_asset_factory(part: str) -> AssetsDefinition:
```

Comment:
Should probably standardize on `profile` or `capacity_factors` or some other name that can remain consistent throughout the whole process.

Reply:
The structure of the thermal derate data is the same, but it's not aggregated, so there's ~100M records spread across ~1000 files, and it won't fit in memory, so I think it'll need to be done in pieces as with the CEMS. But luckily the files are each small, so we can parallelize massively and gracefully if we bring them in. But the factory would need to be modified to allow that kind of processing.

Reply:
Is your suggestion to have some gridpath data written to parquet, but not all tables? I'd suggest standardizing whatever the approach you wind up with is, even if the other tables could just be handled in memory. I'm wary of winding up with three different extraction streams for different tables, doesn't seem optimal. Ideally we'd subclass `Extractor` to make a `Parquet` extraction class that writes CSVs to Parquet files, but perhaps this is out of scope for the budget here.

Reply:
This would be on the output side, not the input.  On the inputs, it's a big zipfile filled with thousands of CSVs that each have ~200K rows, and can easily be processed in parallel without ever using too much memory, but we can't write 300M records to a table in SQLite.

Reply:
i am torn on the naming of `generation_profiles` or `capacity_factors`.... It seems like colloquially people call these profiles but they are just a collection of capacity factors. I lean slightly towards capacity factors. and plus the asset name will have `hourly` in it so it'll be pretty clear what it is

Reply:
I think I agree with you @cmgosnell . Something like `raw_gridpathratk__hourly_solar_capacity_factor` is legible and accurate.